### PR TITLE
feat: support algorithms UI

### LINKED ISSUE #370: [Feature] supplementary algorithms UI
### Feature Description (功能描述)

cec25c771c2540dc3f03d8b598935546

### PR DESCRIPTION
**Pull request by Crayonn**
fix #370

### CODE REVIEW

File: hubble-fe/src/stores/GraphManagementStore/dataAnalyzeStore/dataAnalyzeStore.ts [hunk 1]
```diff
@@ -1346,98 +1443,72 @@ export class DataAnalyzeStore {
...
+    //   | NeighborRankParams
+    //   | KStepNeighbor
+    //   | KHop
+    //   | RadiographicInspection
+    //   | SameNeighbor
+    //   | WeightedShortestPath
+    //   | SingleSourceWeightedShortestPath
+    //   | Jaccard
+    //   | PersonalRank
+    //   | null = null;
```
Comment (javeme) [inline hubble-fe/src/stores/GraphManagementStore/dataAnalyzeStore/dataAnalyzeStore.ts:1462]: unused code?

File: hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx [hunk 2]
```diff
@@ -68,11 +68,27 @@ const GraphPopOver: React.FC<GraphPopOverProps> = observer(
...
+                  ({ id }) => id === dataAnalyzeStore.rightClickedGraphData.id
+                );
+
+                if (isUndefined(node)) {
+                  return;
+                }
+
+                if (node.label === '~undefined') {
+                  Message.info({
+                    content: '该顶点是非法顶点，可能是由悬空边导致',
```
Comment (javeme) [inline hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx:81]: can we translate to English

File: hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx [hunk 3]
```diff
@@ -84,6 +100,36 @@ const GraphPopOver: React.FC<GraphPopOverProps> = observer(
...
+                        .vertices
+                    ) === 0
+                  ) {
+                    if (
+                      isEmpty(
+                        dataAnalyzeStore.visNetwork?.getConnectedNodes(node.id)
+                      )
+                    ) {
+                      Message.info({
+                        content: '不存在邻接点',
```
Comment (javeme) [inline hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx:116]: ditto

File: hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx [hunk 4]
```diff
@@ -84,6 +100,36 @@ const GraphPopOver: React.FC<GraphPopOverProps> = observer(
...
+                        size: 'medium',
+                        showCloseIcon: false,
+                        duration: 1
+                      });
+                    } else {
+                      Message.info({
+                        content: '不存在更多邻接点',
```
Comment (javeme) [inline hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx:123]: ditto

File: hubble-fe/src/components/graph-management/data-analyze/query-result/QueryResult.tsx [hunk 5]
```diff
@@ -63,9 +63,9 @@ const QueryResult: React.FC<QueryResultProps> = observer(
         dynHeightStyle.height = 'calc(100vh - 441px)';
       }
 
-      if (algorithmAnalyzerStore.currentAlgorithm === Algorithm.shortestPath) {
-        dynHeightStyle.height = 'calc(100vh - 555px)';
-      }
+      // if (algorithmAnalyzerStore.currentAlgorithm === Algorithm.shortestPath) {
+      //   dynHeightStyle.height = 'calc(100vh - 555px)';
+      // }
```
Comment (javeme) [inline hubble-fe/src/components/graph-management/data-analyze/query-result/QueryResult.tsx:68]: we can delete unused code

File: hubble-fe/src/components/graph-management/data-analyze/algorithm/CustomPath.tsx [hunk 6]
```diff
@@ -0,0 +1,1113 @@
...
+                                placeholder={t(
+                                  'data-analyze.algorithm-forms.custom-path.placeholder.input-multiple-properties'
+                                )}
+                                // errorLocation="layer"
+                                // errorMessage={
+                                //   algorithmAnalyzerStore
+                                //     .validateCustomPathParmasErrorMessage.steps[
+                                //     ruleIndex
+                                //   ].default_weight
+                                // }
```
Comment (javeme) [inline hubble-fe/src/components/graph-management/data-analyze/algorithm/CustomPath.tsx:819]: unused code?

File: hubble-fe/src/components/graph-management/data-analyze/algorithm/CustomPath.tsx [hunk 7]
```diff
@@ -0,0 +1,1113 @@
...
+                                  );
+                                }}
+                                // originInputProps={{
+                                //   onBlur() {
+                                //     algorithmAnalyzerStore.validateCustomPathRules(
+                                //       'default_weight',
+                                //       ruleIndex
+                                //     );
+                                //   }
+                                // }}
```
Comment (javeme) [inline hubble-fe/src/components/graph-management/data-analyze/algorithm/CustomPath.tsx:842]: unused code?

File: hubble-fe/src/stores/GraphManagementStore/graphManagementStore.ts [hunk 8]
```diff
@@ -528,9 +535,15 @@ function filterParams(originParams: GraphDataConfig): GraphDataConfig {
     if (typeof value !== 'undefined') {
       newParams[key] = originParams[key];
     }
+    // if (value !== null && value !== '' && typeof value !== 'undefined') {
+    //   newParams[key] = originParams[key];
+    // }
```
Comment (javeme) [inline hubble-fe/src/stores/GraphManagementStore/graphManagementStore.ts:540]: unused code?

File: hubble-fe/src/stores/GraphManagementStore/dataAnalyzeStore/dataAnalyzeStore.ts (hunk 1 above)
Comment (Crayonn) [reply hubble-fe/src/stores/GraphManagementStore/dataAnalyzeStore/dataAnalyzeStore.ts]: done

**Review by javeme — APPROVED** [review]

**Review by imbajin — APPROVED** [review]
The new repository url is: https://github.com/apache/incubator-hugegraph-toolchain/tree/master/hugegraph-hubble

File: hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx (hunk 2 above)
Comment (wanganjuan) [reply hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx]: > can we translate to English

and use internationalization

File: hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx (hunk 3 above)
Comment (wanganjuan) [reply hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx]: use internationalization

File: hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx (hunk 2 above)
Comment (imbajin) [reply hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx]: @wanganjuan  review & update it in new [PR](https://github.com/apache/incubator-hugegraph-toolchain/pull/270)

Comment (wanganjuan) [reply hubble-fe/src/components/graph-management/data-analyze/query-result/GraphPopOver.tsx]: > @wanganjuan review & update it in new [PR](https://github.com/apache/incubator-hugegraph-toolchain/pull/270)

ok

**Pull request merged**